### ISSUE TITLE
Optimize initial thread pool size

### ISSUE THREAD

**Issue by JochenHiller** [jupnp/jupnp#43]
When using the configurations (default, OSGi) the core pool size and maximum pool size are the same. The default value is 200.
This leads to this inefficient behavior:
* during startup the pool size will be 200, 200 threads will be created
* due to allowCoreThreadTimeout() the core threads will be terminated very fast

It is much more efficient to start with a much lower pool size (e.g. 3). If more threads are needed they will be created by the pool itself.
I observed that inefficiency running jUPnP in an embedded environment (like Rasp Pi 1) with restricted resources (CPU, memory).

I will provide a patch for that improvement.

**Comment by kgoderis** [comment]
Mmhhh... Cfr the email I sent you guys a few hours ago (on Sonos), I am actually playing around with the threadpoolsize myself. All I can say is that if the pool is too small, in env with a lot of Sonos devices, this results in a bunch of time-outs, consequently GENA subscriptions that are not setup, and thus device state that is not polled at all, and thus complete misbehaviour of the Sonos binding. I have not looked at the code you provide in this commit - I hope it does not break things ;-)

**Comment by JochenHiller** [comment]
Hi Karel,

On Wed, Jan 13, 2016 at 10:43 PM, Karel Goderis <notifications@github.com> wrote:
> Mmhhh... Cfr the email I sent you guys a few hours ago (on Sonos), I am actually playing around with the threadpoolsize myself. All I can say is that if the pool is too small, in env with a lot of Sonos devices, this results in a bunch of time-outs, consequently GENA subscriptions that are not setup, and thus device state that is not polled at all, and thus complete misbehaviour of the Sonos binding. I have not looked at the code you provide in this commit - I hope it does not break things ;-)

that is the reason why we set the default pool size to 200 some weeks ago.
But we noted that if also the corePoolSize is 200, all 200 threads will be created during startup, and then closed within 10 seconds when there are no executors yet in pool.
You will not see that on a PC, but we have 5-10 seconds for startup on a Rasp Pi 1 like device.

About your concerns: if needed there will be up to 200 threads created. So this does not change the scalability.

Did you made tests with more than 200 threads?

Bye, Jochen

**Comment by kgoderis** [comment]
I have it at 200 in the .cfg. However, I forgot about this parameter when I set up the debug environment, and there it was at 20, and obviously things went horribly wrong. I do not know what the exact number should be, but the binding does roughly about 10 invocations per device in a poll sequence, so, I would x that with the number of devices. I have 25 sonos devices here, but at 200 it seems to work. 

I am also wondering if all upnp service subscribers should do a lot synchronize() on themselves, because I already had the case that a callback of a GENA subscription updated values "too fast" (in casu: the saveState() and restoreState() in the Sonos binding are working erratically if you save the state, but in the meantime the URI playing on the Sonos was changed to something else). e.g. I would do a synchronize(this) on outgoing calls to the Sonos device, as GENA data that is coming in via callbacks.
